lib/matplotlib/axis.py

```diff
@@ -2267,19 +2267,15 @@ def set_inverted(self, inverted):
     def set_default_intervals(self):
         # docstring inherited
         xmin, xmax = 0., 1.
-        dataMutated = self.axes.dataLim.mutatedx()
-        viewMutated = self.axes.viewLim.mutatedx()
-        if not dataMutated or not viewMutated:
+        # only change view if dataLim has not changed:
+        if not self.axes.dataLim.mutatedx():
             if self.converter is not None:
                 info = self.converter.axisinfo(self.units, self)
                 if info.default_limits is not None:
                     valmin, valmax = info.default_limits
                     xmin = self.converter.convert(valmin, self.units, self)
                     xmax = self.converter.convert(valmax, self.units, self)
-            if not dataMutated:
```

Comment:
Just because we changed the view limits doesn't mean we should change the data limits, which is what this line does...

Reply:
I think this is likely the correct fix, thanks for figuring that out! :-)